server/connection_handler.go

```diff
@@ -723,6 +728,17 @@ func (h *ConnectionHandler) handleCopyDone(_ *pgproto3.CopyDone) (stop bool, end
 		return false, false, err
 	}
 
+	// If we aren't in an explicit/user managed transaction, we need to commit the transaction
+	if !sqlCtx.GetIgnoreAutoCommit() {
```

Comment:
Seems like this check may not properly handle the distinction between explicit/user manage transactions? Perhaps we should persist some kind of state on the connection handler that states whether we're in a COPY automatic transaction.

Reply:
The name `GetIgnoreAutoCommit` is a bit non-intuitive, but this is what GMS sets when an explicit/user managed transaction is encountered. (i.e. `START TRANSACTION` calls `SetIgnoreAutocommit(true)` and then `COMMIT` and `ROLLBACK` clear it). So, I'm pretty sure this is the correct property to use to see if we're in an automatic or an explicitly-managed transaction, but the name is admittedly a bit non-obvious about how GMS uses this property. We could improve this name in GMS fairly easily – it's only used in a small handful of places.